src/api/MainSolver.cc

```diff
@@ -334,15 +335,17 @@ sstat MainSolver::solve()
 
 std::unique_ptr<SimpSMTSolver> MainSolver::createInnerSolver(SMTConfig & config, THandler & thandler) {
     SimpSMTSolver* solver = nullptr;
-    if (config.sat_pure_lookahead())
+    if (config.sat_pure_lookahead()) {
         solver = new LookaheadSMTSolver(config, thandler);
-    else if (config.sat_lookahead_split())
+    } else if (config.sat_lookahead_split()) {
         solver = new LookaheadSplitter(config, thandler);
-    else if (config.use_ghost_vars())
+    } else if (config.use_ghost_vars()) {
         solver = new GhostSMTSolver(config, thandler);
-    else
+    } else if (config.sat_split_type() == spt_scatter) {
+        solver = new ScatterSplitter(config, thandler);
```

Comment:
Shall I remove createInnerSolver in MainSplitter which initialising splitters?

https://github.com/MasoudAsadzade/opensmt-verification/blob/ec1e8609d9c33b5a84ba37e918f95bb179e79390/src/api/MainSplitter.cc#L61

Reply:
No, I think that's good.  I'll commit a suggestion.